modules/core/src/smithy4s/http/Metadata.scala

```diff
@@ -243,9 +246,21 @@ object Metadata {
     def apply[A](implicit instance: Encoder[A]): Encoder[A] = instance
 
     def fromSchema[A](schema: Schema[A]): Encoder[A] = {
+      val statusCodeVisitor = new HttpResponseCodeSchemaVisitor().apply(schema)
       SchemaVisitorMetadataWriter(schema) match {
-        case StructureMetaEncode(f) => (a: A) => f(a)
-        case _                      => (_: A) => Metadata.empty
+        case StructureMetaEncode(f) => { (a: A) =>
+          val struct = f(a)
+          val statusCode = statusCodeVisitor match {
```

Comment:
same comment as before: you could extract this pattern matching outside of the scope where `A` is known  and have a `A => Option[Int]`.